pkg/evaluator/expression_test.go

```diff
@@ -100,6 +100,67 @@ func TestEvaluate(t *testing.T) {
 			},
 			expectedResult: true,
 		},
+		{
+			expression: "!($appeal.role contains 'viewer') || $appeal.resource.provider_type == 'gcloud_iam' && $appeal.options.duration in ['0h','']",
+			params: map[string]interface{}{
+				"appeal": map[string]interface{}{
+					"role": "Editor",
+					"resource": map[string]interface{}{
+						"provider_type": "gcloud_iam",
+					},
+					"options": map[string]interface{}{
+						"duration": "0h",
+					},
+				},
+			},
+			expectedResult: true,
+		},
+		{
+			expression: "!($appeal.role contains 'viewer') || $appeal.resource.provider_type == 'gcloud_iam' && $appeal.options.duration in ['0h',''] && $appeal.role contains 'viewer'",
+			params: map[string]interface{}{
+				"appeal": map[string]interface{}{
+					"role": "viewer",
+					"resource": map[string]interface{}{
+						"provider_type": "gcloud_iam",
+					},
+					"options": map[string]interface{}{
+						"duration": "0h",
+					},
+				},
+			},
+			expectedResult: true,
+		},
+		{
+			expression: "!($appeal.role contains 'viewer') || $appeal.resource.provider_type == 'gcloud_iam' && $appeal.options.duration in ['0h',''] && $appeal.role contains 'viewer'",
+			params: map[string]interface{}{
+				"appeal": map[string]interface{}{
+					"role": "viewer",
+					"resource": map[string]interface{}{
+						"provider_type": "gcloud_iam",
+					},
+					"options": map[string]interface{}{
+						"duration": "2160h",
+					},
+				},
+			},
+			expectedResult: false,
+		},
+		{
+			expression: "!($appeal.role contains 'admin_user') || $appeal.resource.provider_type == 'gcloud_iam' && $appeal.options.duration in ['0h',''] && $appeal.role contains 'admin_user'",
```

Comment:
need help to add `name` or `description` to the test cases to help understand the scenario

Reply:
Other existing test cases also don't have these as most of the existing ones are straightforward. 
Is it okay if I add comments for these new test cases?

Reply:
yeah works

Reply:
updated